Algorithms/Fibonacci Retracement.py

```diff
@@ -0,0 +1,86 @@
+#Fibonacci Retracement
+from PythonDataProcessing import DataRetrieval as DR
+from numpy import NaN
+
+
+data = DR.get_stock_data('DIS', period='1y')
+
+
+
+
+
+
+
+
+
+def add_fibonacci_retracement(data):
+    '''
+    This function takes a data frame of historic data and finds the fibonacci values for an interval of time and adds the values to the data frame.
+
+    Input:
+        *data* (required) - The dataframe of historic data the fibonacci levels will be found from.
+    
+    Output:
+        *data* - A data frame with the fibonacci levels added
+    
+    Examples:
+        add_fibonacci_retracement(data_source)
+    '''
+
+    #Currently an interval of 50
+    fib_interval = 50
+    i = 0
+
+    fib_interval = fib_interval -1
+
+    FRL = [NaN] *fib_interval
+    FR236 = [NaN] *fib_interval
+    FR382 = [NaN] *fib_interval
+    FR5 = [NaN] *fib_interval
+    FR618 = [NaN] *fib_interval
+    FR786 = [NaN] *fib_interval
+    FRH = [NaN] *fib_interval
+
+    fib_interval = fib_interval +1
+
+    while i+fib_interval <= data.shape[0]:

```

Comment:
I would challenge you to try to do this without this loop using a little bit more list comprehension. 

Try creating a min/max list where each row corresponds to each row in the data. Something like this:

```mn_lst = [min(data[I-fib_interval: fib_interval] for i in range(fib_interval,len(data))]```